staging/src/k8s.io/component-base/featuregate/feature_gate_test.go

```diff
@@ -207,16 +209,49 @@ func TestFeatureGateFlag(t *testing.T) {
 		},
 	}
 	for i, test := range tests {
-		t.Run(test.arg, func(t *testing.T) {
+		t.Run(fmt.Sprintf("AddFlag/%s", test.arg), func(t *testing.T) {
 			fs := pflag.NewFlagSet("testfeaturegateflag", pflag.ContinueOnError)
 			f := NewFeatureGate()
-			f.Add(map[Feature]FeatureSpec{
+			err := f.Add(map[Feature]FeatureSpec{
 				testAlphaGate: {Default: false, PreRelease: Alpha},
 				testBetaGate:  {Default: false, PreRelease: Beta},
 			})
+			if err != nil {
+				t.Errorf("%d: f.Add() Expected nil, Got %v", i, err)
+			}
 			f.AddFlag(fs)
 
-			err := fs.Parse([]string{fmt.Sprintf("--%s=%s", flagName, test.arg)})
+			err = fs.Parse([]string{fmt.Sprintf("--%s=%s", flagName, test.arg)})
+			if test.parseError != "" {
+				if !strings.Contains(err.Error(), test.parseError) {
+					t.Errorf("%d: Parse() Expected %v, Got %v", i, test.parseError, err)
+				}
+			} else if err != nil {
+				t.Errorf("%d: Parse() Expected nil, Got %v", i, err)
+			}
+			for k, v := range test.expect {
+				if actual := f.enabled.Load().(map[Feature]bool)[k]; actual != v {
+					t.Errorf("%d: expected %s=%v, Got %v", i, k, v, actual)
+				}
+			}
+		})
+	}
+	for i, test := range tests {
+		t.Run(fmt.Sprintf("AddGoFlag/%s", test.arg), func(t *testing.T) {
+			var output bytes.Buffer
+			fs := flag.NewFlagSet("testfeaturegateflag", flag.ContinueOnError)
+			fs.SetOutput(&output)
+			f := NewFeatureGate()
+			err := f.Add(map[Feature]FeatureSpec{
+				testAlphaGate: {Default: false, PreRelease: Alpha},
+				testBetaGate:  {Default: false, PreRelease: Beta},
+			})
+			if err != nil {
+				t.Errorf("%d: f.Add() Expected nil, Got %v", i, err)
+			}
+			f.AddGoFlag(fs)
+
+			err = fs.Parse([]string{fmt.Sprintf("-%s", flagName), test.arg})
```

Comment:
I don't like that `feature_gate_test.go` is using internals of the package like `f.enabled.Load().(map[Feature]bool)[k]` instead of `f.Enabled(k)`. This is unrelated to this PR, but perhaps we want to fix it together?

Here's a full patch against master for `git am` which converts this file:
```patch
From c9e4dc5c9e7a08dc4ef122e6df595f7fa763c1fd Mon Sep 17 00:00:00 2001
From: Patrick Ohly <patrick.ohly@intel.com>
Date: Wed, 24 Apr 2024 14:14:12 +0200
Subject: [PATCH] featuregate: black-box unit test

It's better to test the public API of the package without relying on internals.

The way it was done before, major API breaks like renaming the command line
name or the special strings for "all alpha/beta" would not have been caught by
the unit test.
---
 .../featuregate/feature_gate_test.go          | 283 +++++++++---------
 1 file changed, 142 insertions(+), 141 deletions(-)

diff --git a/staging/src/k8s.io/component-base/featuregate/feature_gate_test.go b/staging/src/k8s.io/component-base/featuregate/feature_gate_test.go
index 063a5a36368..90d59403f57 100644
--- a/staging/src/k8s.io/component-base/featuregate/feature_gate_test.go
+++ b/staging/src/k8s.io/component-base/featuregate/feature_gate_test.go
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package featuregate
+package featuregate_test
 
 import (
 	"fmt"
@@ -24,6 +24,7 @@ import (
 	"github.com/spf13/pflag"
 	"github.com/stretchr/testify/assert"
 
+	"k8s.io/component-base/featuregate"
 	"k8s.io/component-base/metrics/legacyregistry"
 	featuremetrics "k8s.io/component-base/metrics/prometheus/feature"
 	"k8s.io/component-base/metrics/testutil"
@@ -31,28 +32,28 @@ import (
 
 func TestFeatureGateFlag(t *testing.T) {
 	// gates for testing
-	const testAlphaGate Feature = "TestAlpha"
-	const testBetaGate Feature = "TestBeta"
+	const testAlphaGate featuregate.Feature = "TestAlpha"
+	const testBetaGate featuregate.Feature = "TestBeta"
 
 	tests := []struct {
 		arg        string
-		expect     map[Feature]bool
+		expect     map[featuregate.Feature]bool
 		parseError string
 	}{
 		{
 			arg: "",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "fooBarBaz=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -60,27 +61,27 @@ func TestFeatureGateFlag(t *testing.T) {
 		},
 		{
 			arg: "AllAlpha=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "AllAlpha=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  true,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    true,
+				"AllBeta":     false,
 				testAlphaGate: true,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "AllAlpha=banana",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -88,45 +89,45 @@ func TestFeatureGateFlag(t *testing.T) {
 		},
 		{
 			arg: "AllAlpha=false,TestAlpha=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: true,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "TestAlpha=true,AllAlpha=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: true,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "AllAlpha=true,TestAlpha=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  true,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    true,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "TestAlpha=false,AllAlpha=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  true,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    true,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "TestBeta=true,AllAlpha=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  true,
 			},
@@ -134,27 +135,27 @@ func TestFeatureGateFlag(t *testing.T) {
 
 		{
 			arg: "AllBeta=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "AllBeta=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   true,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     true,
 				testAlphaGate: false,
 				testBetaGate:  true,
 			},
 		},
 		{
 			arg: "AllBeta=banana",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -162,45 +163,45 @@ func TestFeatureGateFlag(t *testing.T) {
 		},
 		{
 			arg: "AllBeta=false,TestBeta=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  true,
 			},
 		},
 		{
 			arg: "TestBeta=true,AllBeta=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: false,
 				testBetaGate:  true,
 			},
 		},
 		{
 			arg: "AllBeta=true,TestBeta=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   true,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     true,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "TestBeta=false,AllBeta=true",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   true,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     true,
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
 		},
 		{
 			arg: "TestAlpha=true,AllBeta=false",
-			expect: map[Feature]bool{
-				allAlphaGate:  false,
-				allBetaGate:   false,
+			expect: map[featuregate.Feature]bool{
+				"AllAlpha":    false,
+				"AllBeta":     false,
 				testAlphaGate: true,
 				testBetaGate:  false,
 			},
@@ -209,14 +210,14 @@ func TestFeatureGateFlag(t *testing.T) {
 	for i, test := range tests {
 		t.Run(test.arg, func(t *testing.T) {
 			fs := pflag.NewFlagSet("testfeaturegateflag", pflag.ContinueOnError)
-			f := NewFeatureGate()
-			f.Add(map[Feature]FeatureSpec{
-				testAlphaGate: {Default: false, PreRelease: Alpha},
-				testBetaGate:  {Default: false, PreRelease: Beta},
+			f := featuregate.NewFeatureGate()
+			f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
+				testAlphaGate: {Default: false, PreRelease: featuregate.Alpha},
+				testBetaGate:  {Default: false, PreRelease: featuregate.Beta},
 			})
 			f.AddFlag(fs)
 
-			err := fs.Parse([]string{fmt.Sprintf("--%s=%s", flagName, test.arg)})
+			err := fs.Parse([]string{fmt.Sprintf("--feature-gates=%s", test.arg)})
 			if test.parseError != "" {
 				if !strings.Contains(err.Error(), test.parseError) {
 					t.Errorf("%d: Parse() Expected %v, Got %v", i, test.parseError, err)
@@ -225,7 +226,7 @@ func TestFeatureGateFlag(t *testing.T) {
 				t.Errorf("%d: Parse() Expected nil, Got %v", i, err)
 			}
 			for k, v := range test.expect {
-				if actual := f.enabled.Load().(map[Feature]bool)[k]; actual != v {
+				if actual := f.Enabled(k); actual != v {
 					t.Errorf("%d: expected %s=%v, Got %v", i, k, v, actual)
 				}
 			}
@@ -234,14 +235,14 @@ func TestFeatureGateFlag(t *testing.T) {
 }
 
 func TestFeatureGateOverride(t *testing.T) {
-	const testAlphaGate Feature = "TestAlpha"
-	const testBetaGate Feature = "TestBeta"
+	const testAlphaGate featuregate.Feature = "TestAlpha"
+	const testBetaGate featuregate.Feature = "TestBeta"
 
 	// Don't parse the flag, assert defaults are used.
-	var f *featureGate = NewFeatureGate()
-	f.Add(map[Feature]FeatureSpec{
-		testAlphaGate: {Default: false, PreRelease: Alpha},
-		testBetaGate:  {Default: false, PreRelease: Beta},
+	f := featuregate.NewFeatureGate()
+	f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
+		testAlphaGate: {Default: false, PreRelease: featuregate.Alpha},
+		testBetaGate:  {Default: false, PreRelease: featuregate.Beta},
 	})
 
 	f.Set("TestAlpha=true,TestBeta=true")
@@ -263,14 +264,14 @@ func TestFeatureGateOverride(t *testing.T) {
 
 func TestFeatureGateFlagDefaults(t *testing.T) {
 	// gates for testing
-	const testAlphaGate Feature = "TestAlpha"
-	const testBetaGate Feature = "TestBeta"
+	const testAlphaGate featuregate.Feature = "TestAlpha"
+	const testBetaGate featuregate.Feature = "TestBeta"
 
 	// Don't parse the flag, assert defaults are used.
-	var f *featureGate = NewFeatureGate()
-	f.Add(map[Feature]FeatureSpec{
-		testAlphaGate: {Default: false, PreRelease: Alpha},
-		testBetaGate:  {Default: true, PreRelease: Beta},
+	f := featuregate.NewFeatureGate()
+	f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
+		testAlphaGate: {Default: false, PreRelease: featuregate.Alpha},
+		testBetaGate:  {Default: true, PreRelease: featuregate.Beta},
 	})
 
 	if f.Enabled(testAlphaGate) != false {
@@ -284,19 +285,19 @@ func TestFeatureGateFlagDefaults(t *testing.T) {
 func TestFeatureGateKnownFeatures(t *testing.T) {
 	// gates for testing
 	const (
-		testAlphaGate      Feature = "TestAlpha"
-		testBetaGate       Feature = "TestBeta"
-		testGAGate         Feature = "TestGA"
-		testDeprecatedGate Feature = "TestDeprecated"
+		testAlphaGate      featuregate.Feature = "TestAlpha"
+		testBetaGate       featuregate.Feature = "TestBeta"
+		testGAGate         featuregate.Feature = "TestGA"
+		testDeprecatedGate featuregate.Feature = "TestDeprecated"
 	)
 
 	// Don't parse the flag, assert defaults are used.
-	var f *featureGate = NewFeatureGate()
-	f.Add(map[Feature]FeatureSpec{
-		testAlphaGate:      {Default: false, PreRelease: Alpha},
-		testBetaGate:       {Default: true, PreRelease: Beta},
-		testGAGate:         {Default: true, PreRelease: GA},
-		testDeprecatedGate: {Default: false, PreRelease: Deprecated},
+	f := featuregate.NewFeatureGate()
+	f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
+		testAlphaGate:      {Default: false, PreRelease: featuregate.Alpha},
+		testBetaGate:       {Default: true, PreRelease: featuregate.Beta},
+		testGAGate:         {Default: true, PreRelease: featuregate.GA},
+		testDeprecatedGate: {Default: false, PreRelease: featuregate.Deprecated},
 	})
 
 	known := strings.Join(f.KnownFeatures(), " ")
@@ -309,15 +310,15 @@ func TestFeatureGateKnownFeatures(t *testing.T) {
 
 func TestFeatureGateSetFromMap(t *testing.T) {
 	// gates for testing
-	const testAlphaGate Feature = "TestAlpha"
-	const testBetaGate Feature = "TestBeta"
-	const testLockedTrueGate Feature = "TestLockedTrue"
-	const testLockedFalseGate Feature = "TestLockedFalse"
+	const testAlphaGate featuregate.Feature = "TestAlpha"
+	const testBetaGate featuregate.Feature = "TestBeta"
+	const testLockedTrueGate featuregate.Feature = "TestLockedTrue"
+	const testLockedFalseGate featuregate.Feature = "TestLockedFalse"
 
 	tests := []struct {
 		name        string
 		setmap      map[string]bool
-		expect      map[Feature]bool
+		expect      map[featuregate.Feature]bool
 		setmapError string
 	}{
 		{
@@ -326,7 +327,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 				"TestAlpha": true,
 				"TestBeta":  true,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: true,
 				testBetaGate:  true,
 			},
@@ -336,7 +337,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 			setmap: map[string]bool{
 				"TestBeta": true,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: false,
 				testBetaGate:  true,
 			},
@@ -346,7 +347,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 			setmap: map[string]bool{
 				"TestAlpha": false,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -356,7 +357,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 			setmap: map[string]bool{
 				"TestInvaild": true,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -368,7 +369,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 				"TestLockedTrue":  true,
 				"TestLockedFalse": false,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -378,7 +379,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 			setmap: map[string]bool{
 				"TestLockedTrue": false,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -389,7 +390,7 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 			setmap: map[string]bool{
 				"TestLockedFalse": true,
 			},
-			expect: map[Feature]bool{
+			expect: map[featuregate.Feature]bool{
 				testAlphaGate: false,
 				testBetaGate:  false,
 			},
@@ -398,12 +399,12 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 	}
 	for i, test := range tests {
 		t.Run(fmt.Sprintf("SetFromMap %s", test.name), func(t *testing.T) {
-			f := NewFeatureGate()
-			f.Add(map[Feature]FeatureSpec{
-				testAlphaGate:       {Default: false, PreRelease: Alpha},
-				testBetaGate:        {Default: false, PreRelease: Beta},
-				testLockedTrueGate:  {Default: true, PreRelease: GA, LockToDefault: true},
-				testLockedFalseGate: {Default: false, PreRelease: GA, LockToDefault: true},
+			f := featuregate.NewFeatureGate()
+			f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
+				testAlphaGate:       {Default: false, PreRelease: featuregate.Alpha},
+				testBetaGate:        {Default: false, PreRelease: featuregate.Beta},
+				testLockedTrueGate:  {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
+				testLockedFalseGate: {Default: false, PreRelease: featuregate.GA, LockToDefault: true},
 			})
 			err := f.SetFromMap(test.setmap)
 			if test.setmapError != "" {
@@ -427,10 +428,10 @@ func TestFeatureGateSetFromMap(t *testing.T) {
 func TestFeatureGateMetrics(t *testing.T) {
 	// gates for testing
 	featuremetrics.ResetFeatureInfoMetric()
-	const testAlphaGate Feature = "TestAlpha"
-	const testBetaGate Feature = "TestBeta"
-	const testAlphaEnabled Feature = "TestAlphaEnabled"
-	const testBetaDisabled Feature = "TestBetaDisabled"
+	const testAlphaGate featuregate.Feature = "TestAlpha"
+	const testBetaGate featuregate.Feature = "TestBeta"
+	const testAlphaEnabled featuregate.Feature = "TestAlphaEnabled"
+	const testBetaDisabled featuregate.Feature = "TestBetaDisabled"
 	testedMetrics := []string{"kubernetes_feature_enabled"}
 	expectedOutput := `
 		# HELP kubernetes_feature_enabled [BETA] This metric records the data about the stage and enablement of a k8s feature.
@@ -443,12 +444,12 @@ func TestFeatureGateMetrics(t *testing.T) {
 		kubernetes_feature_enabled{name="TestBetaDisabled",stage="ALPHA"} 0
 `
 
-	f := NewFeatureGate()
-	fMap := map[Feature]FeatureSpec{
-		testAlphaGate:    {Default: false, PreRelease: Alpha},
-		testAlphaEnabled: {Default: false, PreRelease: Alpha},
-		testBetaGate:     {Default: true, PreRelease: Beta},
-		testBetaDisabled: {Default: true, PreRelease: Alpha},
+	f := featuregate.NewFeatureGate()
+	fMap := map[featuregate.Feature]featuregate.FeatureSpec{
+		testAlphaGate:    {Default: false, PreRelease: featuregate.Alpha},
+		testAlphaEnabled: {Default: false, PreRelease: featuregate.Alpha},
+		testBetaGate:     {Default: true, PreRelease: featuregate.Beta},
+		testBetaDisabled: {Default: true, PreRelease: featuregate.Alpha},
 	}
 	f.Add(fMap)
 	f.SetFromMap(map[string]bool{"TestAlphaEnabled": true, "TestBetaDisabled": false})
@@ -460,14 +461,14 @@ func TestFeatureGateMetrics(t *testing.T) {
 
 func TestFeatureGateString(t *testing.T) {
 	// gates for testing
-	const testAlphaGate Feature = "TestAlpha"
-	const testBetaGate Feature = "TestBeta"
-	const testGAGate Feature = "TestGA"
-
-	featuremap := map[Feature]FeatureSpec{
-		testGAGate:    {Default: true, PreRelease: GA},
-		testAlphaGate: {Default: false, PreRelease: Alpha},
-		testBetaGate:  {Default: true, PreRelease: Beta},
+	const testAlphaGate featuregate.Feature = "TestAlpha"
+	const testBetaGate featuregate.Feature = "TestBeta"
+	const testGAGate featuregate.Feature = "TestGA"
+
+	featuremap := map[featuregate.Feature]featuregate.FeatureSpec{
+		testGAGate:    {Default: true, PreRelease: featuregate.GA},
+		testAlphaGate: {Default: false, PreRelease: featuregate.Alpha},
+		testBetaGate:  {Default: true, PreRelease: featuregate.Beta},
 	}
 
 	tests := []struct {
@@ -498,7 +499,7 @@ func TestFeatureGateString(t *testing.T) {
 	}
 	for i, test := range tests {
 		t.Run(fmt.Sprintf("SetFromMap %s", test.expect), func(t *testing.T) {
-			f := NewFeatureGate()
+			f := featuregate.NewFeatureGate()
 			f.Add(featuremap)
 			f.SetFromMap(test.setmap)
 			result := f.String()
@@ -511,8 +512,8 @@ func TestFeatureGateString(t *testing.T) {
 
 func TestFeatureGateOverrideDefault(t *testing.T) {
 	t.Run("overrides take effect", func(t *testing.T) {
-		f := NewFeatureGate()
-		if err := f.Add(map[Feature]FeatureSpec{
+		f := featuregate.NewFeatureGate()
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
 			"TestFeature1": {Default: true},
 			"TestFeature2": {Default: false},
 		}); err != nil {
@@ -533,8 +534,8 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 	})
 
 	t.Run("overrides are preserved across deep copies", func(t *testing.T) {
-		f := NewFeatureGate()
-		if err := f.Add(map[Feature]FeatureSpec{"TestFeature": {Default: false}}); err != nil {
+		f := featuregate.NewFeatureGate()
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{"TestFeature": {Default: false}}); err != nil {
 			t.Fatal(err)
 		}
 		if err := f.OverrideDefault("TestFeature", true); err != nil {
@@ -547,10 +548,10 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 	})
 
 	t.Run("reflected in known features", func(t *testing.T) {
-		f := NewFeatureGate()
-		if err := f.Add(map[Feature]FeatureSpec{"TestFeature": {
+		f := featuregate.NewFeatureGate()
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{"TestFeature": {
 			Default:    false,
-			PreRelease: Alpha,
+			PreRelease: featuregate.Alpha,
 		}}); err != nil {
 			t.Fatal(err)
 		}
@@ -573,8 +574,8 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 	})
 
 	t.Run("may not change default for specs with locked defaults", func(t *testing.T) {
-		f := NewFeatureGate()
-		if err := f.Add(map[Feature]FeatureSpec{
+		f := featuregate.NewFeatureGate()
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
 			"LockedFeature": {
 				Default:       true,
 				LockToDefault: true,
@@ -591,8 +592,8 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 	})
 
 	t.Run("does not supersede explicitly-set value", func(t *testing.T) {
-		f := NewFeatureGate()
-		if err := f.Add(map[Feature]FeatureSpec{"TestFeature": {Default: true}}); err != nil {
+		f := featuregate.NewFeatureGate()
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{"TestFeature": {Default: true}}); err != nil {
 			t.Fatal(err)
 		}
 		if err := f.OverrideDefault("TestFeature", false); err != nil {
@@ -607,11 +608,11 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 	})
 
 	t.Run("prevents re-registration of feature spec after overriding default", func(t *testing.T) {
-		f := NewFeatureGate()
-		if err := f.Add(map[Feature]FeatureSpec{
+		f := featuregate.NewFeatureGate()
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
 			"TestFeature": {
 				Default:    true,
-				PreRelease: Alpha,
+				PreRelease: featuregate.Alpha,
 			},
 		}); err != nil {
 			t.Fatal(err)
@@ -619,10 +620,10 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 		if err := f.OverrideDefault("TestFeature", false); err != nil {
 			t.Fatal(err)
 		}
-		if err := f.Add(map[Feature]FeatureSpec{
+		if err := f.Add(map[featuregate.Feature]featuregate.FeatureSpec{
 			"TestFeature": {
 				Default:    true,
-				PreRelease: Alpha,
+				PreRelease: featuregate.Alpha,
 			},
 		}); err == nil {
 			t.Error("expected re-registration to return a non-nil error after overriding its default")
@@ -630,14 +631,14 @@ func TestFeatureGateOverrideDefault(t *testing.T) {
 	})
 
 	t.Run("does not allow override for an unknown feature", func(t *testing.T) {
-		f := NewFeatureGate()
+		f := featuregate.NewFeatureGate()
 		if err := f.OverrideDefault("TestFeature", true); err == nil {
 			t.Error("expected an error to be returned in attempt to override default for unregistered feature")
 		}
 	})
 
 	t.Run("returns error if already added to flag set", func(t *testing.T) {
-		f := NewFeatureGate()
+		f := featuregate.NewFeatureGate()
 		fs := pflag.NewFlagSet("test", pflag.ContinueOnError)
 		f.AddFlag(fs)
 
-- 
2.39.2

```